File: kotlin/src/main/kotlin/advertisement/domain/exceptions/DescriptionEmptyException.kt
package advertisement.domain.exceptions

class DescriptionEmptyException private constructor(message: String) : Exception(message) {

    companion object {
        fun build(): DescriptionEmptyException {
            return DescriptionEmptyException("Description empty")
        }
    }

    override fun fillInStackTrace(): Throwable {
        if (shouldIncludeStackTrace()) {
            return super.fillInStackTrace()
        }
        return this
    }

    private fun shouldIncludeStackTrace(): Boolean {
        // Implement logic to determine if stack trace is needed (e.g., based on configuration)
        return false
    }
}

File: typescript/src/advertisement/domain/exceptions/DescriptionEmptyException.ts
import {DomainException} from "../../../common/domain/DomainException";

export class DescriptionEmptyException extends DomainException {
  private static _message: string = 'Description empty';

  private constructor(message?: string) {
    super(message);
    this.name = 'DescriptionEmptyException';
    Object.setPrototypeOf(this, new.target.prototype);
  }

  static build(): DescriptionEmptyException {
    return new DescriptionEmptyException(this._message);
  }
}
